### DISCUSSION TITLE
How to redirect to route in an action?

### DISCUSSION THREAD

**Discussion by OvalMedia** [filamentphp/filament#5464] [Help] [answered]
I have an action with a custom route to a controller that updates a record:

```
Actions\Action::make('invoice_pay')
	->label('pay')
	->color('success')
	->icon('bxs-coin-stack')
	->url(route('invoices.pay', ['invoice' => $this->data['id']]))
```

This works fine so far. Now i want a modal confirmation before the route is called:

```
Actions\Action::make('invoice_pay')
	->label('pay')
	->color('success')
	->icon('bxs-coin-stack')
	->requiresConfirmation(true)
	->url(route('invoices.pay', ['invoice' => $this->data['id']]))
```

But the modal never appears. 
In the [docs](https://filamentphp.com/docs/2.x/admin/pages/actions#confirmation-modals) it says that confirmation modals don't work when an url() is set, I should use a redirect in the action method instead. But the docs fail to tell me how to do it.

```
Actions\Action::make('invoice_pay')
	->label('pay')
	->color('success')
	->icon('bxs-coin-stack')
	->requiresConfirmation(true)
	->action(redirect()->route('invoices.pay', ['invoice' => $this->data['id']]))
```

This results in an error:

`Filament\Support\Actions\Action::action(): Argument #1 ($action) must be of type Closure|string|null, Livewire\Redirector given, called in .../app/Filament/Resources/InvoiceResource/Pages/EditInvoice.php on line 38`

**Comment by OvalMedia** [comment, marked as answer]
Never mind, found it: `->action(fn () => redirect()->route('invoices.pay', ['invoice' => $this->data['id']]))`

**Reply by OvalMedia** [reply]
Still: it wouldn't hurt if the docs would supply an example, it's a one-liner after all.

**Reply by danharrin** [reply]
We didn't add it because its kinda a Livewire thing?

**Reply by OvalMedia** [reply]
A newbie like myself could not tell. I think the docs could give an example (not only in this case) AND point out that this would be a "livewire thing". Maybe even point to the livewire docs. That way a user can gradually learn about the different puzzle pieces this whole thing is composed of. It may be obvious for you as the developer but it is not necessarily for everybody else.

**Reply by bilogic** [reply]
@OvalMedia I have to agree. Are you using v3 now? Any idea about my issue here https://github.com/filamentphp/filament/discussions/8775 ? Thank you.

**Reply by bilogic** [reply]
I'm back here again because I want the redirect URL to open in a new tab, any ideas?

**Reply by danharrin** [reply]
It isn't possible to redirect to a new tab in Livewire, any attempts through JavaScript (which handles LW redirects) would result in the pop up being blocked by the browser. New tab clicks need to be initiated by the user through an `<a>` tag

**Reply by bilogic** [reply]
ok thanks @danharrin
